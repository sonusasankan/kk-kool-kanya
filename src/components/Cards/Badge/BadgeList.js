import maternityBenifits from '../../../assets/badges/ic-maternity-benefit-28.svg'
import safety from '../../../assets/badges/ic-safety-28.svg';
import salaryBenifits from '../../../assets/badges/ic-salarybenefit-28.svg';
import femaleRepresentatives from '../../../assets/badges/ic-female-representative-28.svg';
import culture from '../../../assets/badges/ic-culture-28.svg';
import learning from '../../../assets/badges/ic-learning-opportunity-28.svg';
import workLifeBalance from '../../../assets/badges/ic-work-life-balance-28.svg';
import workSatisfaction from '../../../assets/badges/ic-job-satisfaction-28.svg';

const BadgeList = {
    MATERNITY_BENEFITS:     maternityBenifits,
    SAFETY:                 safety,
    SALARY_BENEFITS:        salaryBenifits,
    FEMALE_REPRESENTATIVES: femaleRepresentatives,
    CULTURE:                culture,
    LEARNING_OPPORTUNITIES: learning,
    WORKLIFE_BALANCE:       workLifeBalance,
    WORK_SATISFACTION:      workSatisfaction
}

export default BadgeList;