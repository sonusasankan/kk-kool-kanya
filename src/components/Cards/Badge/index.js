export { default } from './Badge';
export {default as BadgeList} from './BadgeList';
