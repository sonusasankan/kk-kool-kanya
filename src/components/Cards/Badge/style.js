import styled from 'styled-components';

export const BadgeWrapper = styled.div`
    ${props => ((props.top || props.right || props.bottom || props.left) ? 'position: absolute' : '')};
    ${props => (props.top ? 'top: 1rem' : '')};
    ${props => (props.right ? 'right: 1rem' : '')};
    ${props => (props.bottom ? 'bottom: 1rem' : '')};
    ${props => (props.left ? 'left: 1rem' : '')};
    z-index: 2;
`;

export const Badge = styled.div`
    display: inline-block;
    height: 24px;
    width: 24px;  
    position: relative;
    img{
        width: 100%;
    }
    margin-left: -10px;
    &:nth-child(1){
        margin-left: 0;
    }
`;
