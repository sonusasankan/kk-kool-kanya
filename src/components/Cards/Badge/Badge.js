import React from 'react';
import PropTypes from 'prop-types';

import { Badge, BadgeWrapper } from './style';

// function createBadges(badgesArr) {
//   badgesArr.map(element => {
//     console.log(element);
//     return (
//       <div>     
//         <img src={element} alt="badge"/>
//       </div>
//     )
//   });
//   return badgesArr;
// }

const badge = ({
  top,
  right,
  bottom,
  left,
  badges,
}) => (
  <BadgeWrapper
    top={top}
    right={right}
    bottom={bottom}
    left={left}
    className="card__badges"
  >
    {badges.map((element, index) => {
      return (
        <Badge className='kk-badge' style={{zIndex : 10 - index}}>
          <img className="img-fluid" src={element} alt="badge" />
        </Badge>
      )
    })}
  </BadgeWrapper>
);

badge.propTypes = {
  top: PropTypes.bool,
  right: PropTypes.bool,
  bottom: PropTypes.bool,
  left: PropTypes.bool,
  numberOfBadges: PropTypes.string.isRequired,
};

badge.defaultProps = {
  top: false,
  right: false,
  bottom: false,
  left: false,
};

export default badge;
