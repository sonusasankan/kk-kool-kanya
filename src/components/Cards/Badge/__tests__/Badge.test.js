import React from 'react';

import { shallow } from 'enzyme';
import { default as CardBadge } from '../Badge';


describe('Badge', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<CardBadge numberOfBadges="3" />);
    expect(component).toMatchSnapshot();
  });
});
