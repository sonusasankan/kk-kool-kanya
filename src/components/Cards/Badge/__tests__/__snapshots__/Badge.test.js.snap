// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <badge
    bottom={false}
    left={false}
    numberOfBadges="3"
    right={false}
    top={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "bottom": false,
      "children": Array [
        <ForwardRef>
          <img
            alt=""
            src="ic-awards-24@2x.png"
          />
        </ForwardRef>,
        <ForwardRef>
          <img
            alt=""
            src="ic-awards-24@2x.png"
          />
        </ForwardRef>,
        <ForwardRef>
          <img
            alt=""
            src="ic-awards-24@2x.png"
          />
        </ForwardRef>,
      ],
      "className": "card__badges",
      "left": false,
      "right": false,
      "top": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <img
            alt=""
            src="ic-awards-24@2x.png"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "ic-awards-24@2x.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
    display: inline-block;
    height: 24px;
    width: 24px;  
    img{
        width: 100%;
    }
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <img
            alt=""
            src="ic-awards-24@2x.png"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "ic-awards-24@2x.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
    display: inline-block;
    height: 24px;
    width: 24px;  
    img{
        width: 100%;
    }
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <img
            alt=""
            src="ic-awards-24@2x.png"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "ic-awards-24@2x.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
    display: inline-block;
    height: 24px;
    width: 24px;  
    img{
        width: 100%;
    }
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    ",
          [Function],
          ";
    ",
          [Function],
          ";
    ",
          [Function],
          ";
    ",
          [Function],
          ";
    ",
          [Function],
          ";
    z-index: 2;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "bottom": false,
        "children": Array [
          <ForwardRef>
            <img
              alt=""
              src="ic-awards-24@2x.png"
            />
          </ForwardRef>,
          <ForwardRef>
            <img
              alt=""
              src="ic-awards-24@2x.png"
            />
          </ForwardRef>,
          <ForwardRef>
            <img
              alt=""
              src="ic-awards-24@2x.png"
            />
          </ForwardRef>,
        ],
        "className": "card__badges",
        "left": false,
        "right": false,
        "top": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <img
              alt=""
              src="ic-awards-24@2x.png"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "src": "ic-awards-24@2x.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
    display: inline-block;
    height: 24px;
    width: 24px;  
    img{
        width: 100%;
    }
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <img
              alt=""
              src="ic-awards-24@2x.png"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "src": "ic-awards-24@2x.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
    display: inline-block;
    height: 24px;
    width: 24px;  
    img{
        width: 100%;
    }
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <img
              alt=""
              src="ic-awards-24@2x.png"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "src": "ic-awards-24@2x.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
    display: inline-block;
    height: 24px;
    width: 24px;  
    img{
        width: 100%;
    }
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
    ",
            [Function],
            ";
    ",
            [Function],
            ";
    ",
            [Function],
            ";
    ",
            [Function],
            ";
    ",
            [Function],
            ";
    z-index: 2;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
